src/main/java/ch/njol/skript/conditions/CondIsInOpenWater.java

```diff
@@ -0,0 +1,67 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerFishEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Is in Open Water")
+@Description("Check whether or not the fish hook is in open water.")
+@Examples({"on fish:",
+		"\tif hook is in open water:",
+		"\t\tsend \"You will catch a shark soon!\""})
+@Events("fishing")
+@Since("INSERT VERSION")
+public class CondIsInOpenWater extends Condition {
+	
+	static {
+		Skript.registerCondition(CondIsInOpenWater.class,
+				"[fish[ing]] hook (is|are) in open water",
+				"[fish[ing]] hook (isn't|is not|aren't|are not) in open water");
+	}
+	
+	@Override
+	@SuppressWarnings("null")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerFishEvent.class)) {
+			Skript.error("The 'hook is in open water' condition can only be used in fish event.");
+			return false;
+		}
+		setNegated(matchedPattern == 1);
+		return true;
+	}
+	
+	@Override
+	public boolean check(Event e) {
+		return isNegated() ^ ((PlayerFishEvent) e).getHook().isInOpenWater();
```

Comment:
I feel this should be a condition for any fishing hook, not just the event value. This'd require a new type for fishing hooks (not sure how it works exactly), but it is way more flexible.

You can make the fishing hook expression optional to make the event value as default replacement.

Reply:
Good suggestion :+1: I will try to register the fishing hook classinfo and make it usable in all these expressions and conditions